### PR TITLE
feat(package): rename package to use @ibm-cloud scope

### PR DESCRIPTION
**Pull request by briangleeson**
## PR summary
Suggested by Phil Adams as part of SDK review:
* change `ibm-continuous-delivery` package name to `continuous-delivery`, and
* move it to `@ibm-cloud` organisation scope
* update README according to the above changes
* add migration section to README
* remove version numbers from README to avoid need for manual updates in future, and possible mismatches
* add public publish setting to package.json, necessary for the org scoped package change

## PR Checklist
Please make sure that your PR fulfills the following requirements:
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

**Comment by briangleeson** [comment]
Reference PR from platform-services: https://github.com/IBM/platform-services-node-sdk/pull/118

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "ibm-continuous-delivery",
+  "name": "@ibm-cloud/continuous-delivery",
```
Comment (padamstx) [inline package.json:2]: The default "access" for a non-scoped project is public, while the default for a scoped package is private. Therefore, you should add this to your package.json:
```
  "publishConfig": {
    "access": "public"
  },
```
Also, just a reminder to make sure that the NPM id being used by your travis build to publish releases has been given the proper authority in the `@ibm-cloud` npm org (you request this by opening an issue per the instructions re: the use of the `@ibm-cloud` scope).

**Review by padamstx — CHANGES_REQUESTED** [review]
The changes look good, but I left a couple of suggestions:
1. (optional) remove project version # from README
2. (required) add the "publishConfig" object to the package.json file

File: README.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 
-# IBM Cloud Continuous Delivery Node.js SDK 0.2.11
+# IBM Cloud Continuous Delivery Node.js SDK 0.3.0
```
[the remark this answers is not in the document]
Comment (briangleeson) [reply README.md:2]: Good idea, no risk of a mismatch, which we ran into recently

File: package.json (hunk 1 above)
Comment (briangleeson) [reply package.json:2]: Thanks, added that to package.json

@omar-albastami tells me that the npm token is for the ids-org user, so we should be good to go on that front

**Review by padamstx — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (padamstx) [reply package.json:2]: @dpopp07 Does the "ids-org" npm user have the authority to publish an `@ibm-cloud`-scoped package?  Just wanted to double check.

Comment (dpopp07) [reply package.json:2]: The username is `idsorg` but yes, they are a part of the org 👍

**Pull request merged**